model-optimizer/extensions/front/tf/ObjectDetectionAPI.py

```diff
@@ -512,19 +515,104 @@ def update_parameter_shape(graph: Graph, match: [SubgraphMatch, None]):
     return initial_input_node_name, parameter_node
 
 
-class ObjectDetectionAPIPreprocessorReplacement(FrontReplacementFromConfigFileSubGraph):
+class ObjectDetectionAPITransformationsStart(FrontReplacementPattern):
     """
-    The class replaces the "Preprocessor" block resizing input image and applying mean/scale values. Only nodes related
-    to applying mean/scaling values are kept.
+    This is a anchor transformation which is used to distinguish TF OD API models related transformations.
     """
-    replacement_id = 'ObjectDetectionAPIPreprocessorReplacement'
+    enabled = True
+
+    def run_after(self):
+        return [CropAndResizeReplacement, FakeQuantWithMinMaxVarsToQuantize]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        pass
+
+
+class ObjectDetectionAPITransformationsFinish(FrontReplacementPattern):
+    """
+    This is a anchor transformation which is used to distinguish TF OD API models related transformations.
+    """
+    enabled = True
+    # cleanup the graph after applying of TF OD API transformations to remove a lot of unconnected nodes to avoid issues
+    # with shape inference
+    force_clean_up = True
 
     def run_before(self):
         # PadTFToPad inserts Transpose ops for Pad ops inside the sub-graph corresponding to DetectionOutput.
         # But the inputs corresponding to padding values is re-used as inputs for newly created Pad node. This input
         # is removed during removing nodes from the DO sub-graph so the first input to Transpose is missing which
         # results in TransposeOrderNormalizer transformation failure.
-        return [Pack, TransposeOrderNormalizer, PadTFToPad]
+        return [Pack, TransposeOrderNormalizer, PadTFToPad, SqueezeAxis, StandaloneConstEraser, TFSliceToSliceReplacer,
+                KerasRNNOutputConcatenation, KerasRNNInputSlicing]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        pass
```

Comment:
I have the same comment since the transformation seems to do nothing.

Reply:
Refer to the comment above.

Reply:
And it calls the graph cleanup which is very important :)